components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2796,33 +2831,51 @@ private String getSearchAttribute(String filterOperation, String attributeValue,
     }
 
     /**
-     * get list of roles that matches the search criteria
-     * @param filterOperation operator value
-     * @param attributeValue search value
-     * @return list of role names
-     * @throws org.wso2.carbon.user.core
-    .UserStoreException
+     * Check whether the filter attribute support filtering with the domain embedded in the attribute value.
+     *
+     * @param attributeName Attribute to filter
+     * @return True if the given attribute support embedding domain
+     */
+    private boolean isDomainEmbeddingSupportedAttribute(String attributeName) {
+
+        return SCIMConstants.UserSchemaConstants.USER_NAME_URI.equalsIgnoreCase(attributeName)
+                || SCIMConstants.CommonSchemaConstants.ID_URI.equalsIgnoreCase(attributeName)
+                || SCIMConstants.UserSchemaConstants.GROUP_URI.equalsIgnoreCase(attributeName)
+                || SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI.equalsIgnoreCase(attributeName);
+    }
+
+    /**
+     * Get list of roles that matches the search criteria.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Search value
+     * @return List of role names
+     * @throws org.wso2.carbon.user.core.UserStoreException Error getting roleNames.
      */
-    private String[] getRoleNames(String filterOperation, String attributeValue) throws org.wso2.carbon.user.core
-            .UserStoreException {
+    private String[] getRoleNames(String attributeName, String filterOperation, String attributeValue)
```

Comment:
Better to update the signature in order below
filterOperation
attributeName
attributeValue

Reply:
To maintain the consistency, parameters were set in the attributeName, filterOperation and attributeValue order.